docs/main.typ

```diff
@@ -0,0 +1,233 @@
+
+= Shortcomings of LoCC
+
+There are multiple problems when relying on LoCC as a sole metric of productivity.
+
+== First problem: Ambiguous LoCC definitions <loccDef>
+
+Firstly, the term itself is ambiguous and subjective to the formatting of the code. One could say that line breaks are just an arbitrary formatting character. In reality, the program could have been written in a single line of code.
+
+See the following ambiguous examples in @ambiguousListings. A few questions arise: should these snippets be counted as three separate lines of code or collapsed into a single line? In terms of actual work, a developer who writes either version is equally productive, yet if we simply count physical lines changed, the author of the first listing would be credited with three times the contribution of the second. This discrepancy shows how formatting alone can skew LoCC-based distance measures, as trivial style differences inflate the perceived distance between revisions and undermine the metric’s reliability.
+
+#include("assets/ambiguous-line-counts.typ")
+
+// TODO: add source of how different tools have ambiguous definitions of lines
+
+
+== Second problem: Rename-detection pitfalls <renamePitfalls>
+
+The LoCC metric can be distorted also by file renames, which may artificially inflate contribution counts, since renaming a file requires little effort but appears as significant line changes. Git supports rename detection using a similarity threshold#footnote[https://git-scm.com/docs/git-log#Documentation/git-log.txt-code-Mltngtcode], comparing deleted and added files to identify likely renames, utilized by tools such as Git Truck. While Git doesn’t exclude renames by default, this tracking can be used to filter them out manually. However, rename detection is inherently ambiguous — at what point should we stop treating a change as a rename and instead count it as having deleted the old file and added a new one?#footnote[https://en.wikipedia.org/wiki/Ship_of_Theseus]
+
+The issue is further exacerbated when developers squash commits, potentially losing rename information. Ultimately, it’s a trade-off between overcounting trivial renames and missing substantial, legitimate contributions.
+
+== Third problem: Automation-driven LoCC spikes <automationDrivenSpikes>
+
+Another case where the LoCC metric falls short is when performing automated actions that affects a vast amount of files and leads to spikes in line changes. Examples of this include running formatting scripts or installing packages, which often leads to a lot of line changes in tracked lock files [ref].
```

Comment:
Kan man lave et argument for at disse ret nemt kan ses bort fra, da config-filer kan ekskluderes, og man kan enforce korrekt formattering når man committer?

Reply:
True, skal nok skrive formatering og linting også